src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs

```diff
@@ -124,4 +142,27 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             }
         }
     }
+
+    public static class HttpClientExtensions
+    {
+        public static Task<HttpResponseMessage> Send(this HttpClient client, bool async, HttpRequestMessage request, HttpCompletionOption completionOption = default, CancellationToken cancellationToken = default)
+        {
+            if (async)
+            {
+                return client.SendAsync(request, completionOption, cancellationToken);
+            }
+            else
+            {
+#if NETCOREAPP
+                // Note that the sync call must be done on a different thread because it blocks until the server replies.
+                // However, the server-side of the request handling is in many cases invoked after the client, thus deadlocking the test.
+                return Task.Run(() => client.Send(request, completionOption, cancellationToken));
+#else
+                // Do just async for framework, since it'll never have the sync API and
+                // it shouldn't even be called due to AsyncBoolValues returning only true.
+                return client.SendAsync(request, completionOption, cancellationToken);
```

Comment:
If the assertion is that this should never be hit, use `Debug.Fail`.

Reply:
Fixed